src/extension/tests/Test_ActionHandler.py

```diff
@@ -214,8 +214,14 @@ def test_telemetry_not_available(self):
         with self.assertRaises(SystemExit) as sys_exit:
             self.action_handler.enable()
 
-        # event folder is set within HandlerEnvironment but the directory path is invalid. i.e. Telemetry setup not valid
-        self.action_handler.ext_env_handler.events_folder = "testfolder"
+        self.action_handler.ext_env_handler.events_folder = events_folder_path_backup
+
+    def test_telemetry_available_events_folder_not_exists(self):
+        # handler actions will continue to execute after logging telemetry not suported message
+        events_folder_path_backup = self.action_handler.ext_env_handler.events_folder
+        
+        # events folder is set within HandlerEnvironment but the directory path is invalid, so Telemetry is setup but events folder wasn't created
+        self.action_handler.ext_env_handler.events_folder = os.path.join(tempfile.mkdtemp(), "testfolder")
```

Comment:
informational to @rane-rajasi also -- the temp folder creation logic sprinkled around the test code results in folders getting created (with files) in the user's temp directory, and they're not being cleaned up by the tests concerned.

Reply:
We need to remove any temp directory created by UTs. All other parts of the UTs, where a temp dir is created, is followed by a cleanup at the end. 

For eg: the shutil.rmtree() call here
![image](https://user-images.githubusercontent.com/56841542/133528161-f113940e-66f2-4755-8d22-0d9f9a3b7996.png)

@kjohn-msft this part should delete all temp directories. I think only when a test fails, the temp directory cleanup is not carried out

Reply:
Most (if not all) existing tests do cleanup test folder creation. I changed this logic to clean up the test folder created.